driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java

```diff
@@ -251,46 +249,25 @@ private boolean handleReplicaSetMemberChanged(final ServerDescription newDescrip
         if (newDescription.getCanonicalAddress() != null
                 && !newDescription.getAddress().equals(new ServerAddress(newDescription.getCanonicalAddress()))
                 && !newDescription.isPrimary()) {
-            if (LOGGER.isInfoEnabled()) {
-                LOGGER.info(format("Canonical address %s does not match server address.  Removing %s from client view of cluster",
-                                   newDescription.getCanonicalAddress(), newDescription.getAddress()));
-            }
+            LOGGER.info(format("Canonical address %s does not match server address.  Removing %s from client view of cluster",
```

Comment:
This will always format a string even if Logger.isInfoEnabled is false. I recall this pattern being added due to the cost of such operations. Is that desired.

SLF4J uses an varargs method eg: `public void info(String format, Object... arguments);` which prevents this and encompasses the check.  Added JAVA-4521

Reply:
In this context it's an unnecessary optimization as it's executing on background threads.  But it's a drive-by change and can be reverted if you prefer.

Reply:
Would rather fix it in JAVA-4521 - it would improve all the code.